### DISCUSSION TITLE
Step-by-step best practices for CAFE5

### DISCUSSION THREAD

**Discussion by adeoliveira86** [hahnlab/CAFE5#167] [General]
Dear authors, developers, community,

How are you? I used the previous version of CAFE (i.e., CAFE4) in my research and I remember following a really nice tutorial with all the necessary/important steps to generated sounded calculations of gene expansion/contractions in a metazoan genome (e.g., filtering orthogroups, comparing models, selecting multiple lambdas). My question would be is there anything similar to the CAFE5 version? I tried to find an specific pipeline in the discussion and I could not, so if my question is redundant I apologise in advance.

I am looking to the available tutorials present together with the CAFE5 installation (Test_data_readMe and tutorial.md) and I am a bit lost with the pipeline. From what I could see the important steps include:

1. Selecting the most appropriate number of gamma categories by:

- Running CAFE5 with increasing number of discrete gamma categories (say K=1 to K=10)
- Selecting the highest likelihood score for a given k

2. Identifying if there is a need to use multiple λ in the CAFE5
- This can be done by looking to the .cafe file, if I am not mistaken (as described in the tutorial.md)

3. Here I am a bit lost. Because I would like to incorporate the error model and also to compare the log-likelihoods of different models. Which steps should be done first? I could see from the Test_data_readMe that the command --simulate is involved into this in conjuction with the rootdist which can be obtained from step 1.

4. The large families (orthogroups) identified by Orthofinder should be also taken into special consideration as the previous versions?

Which proposed pipeline would be the most appropriate to leverage from all the new implementations in CAFE5? From start to finish?

Thanks again for the help,
André

**Comment by benfulton** [comment]
1. The value of k only indicates how finely to discretize your gamma curve. It shouldn't matter too much which one you choose. 
2. There isn't really a good procedure to determine whether multiple lambdas are necessary. (But if you think of one, let us know!) You would use that more to break off a group you already are interested in - say the humans/chimps branch from the example mammals_tree file.
3. The procedure for calculating a p-value would involve simulating, say, 1000 data sets at random using your tree. Then you can look to see where your likelihood falls in that set of likelihoods.
4. I'm not sure what that means.
